libbeat/cmd/export/config.go

```diff
@@ -8,38 +8,41 @@ import (
 	"gopkg.in/yaml.v2"
 
 	"github.com/elastic/beats/libbeat/cmd/instance"
+	"github.com/elastic/beats/libbeat/common/cli"
 )
 
+// GenExportConfigCmd write to stdout the current configuration in the YAML format.
 func GenExportConfigCmd(name, idxPrefix, beatVersion string) *cobra.Command {
 	return &cobra.Command{
 		Use:   "config",
 		Short: "Export current config to stdout",
-		Run: func(cmd *cobra.Command, args []string) {
-			b, err := instance.NewBeat(name, idxPrefix, beatVersion)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
-				os.Exit(1)
-			}
-
-			err = b.Init()
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error initializing beat: %s\n", err)
-				os.Exit(1)
-			}
-
-			var config map[string]interface{}
-			err = b.RawConfig.Unpack(&config)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error unpacking config")
-				os.Exit(1)
-			}
-			res, err := yaml.Marshal(config)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error converting config to YAML format")
-				os.Exit(1)
-			}
-
-			os.Stdout.Write(res)
-		},
+		Run: cli.RunWith(func(cmd *cobra.Command, args []string) error {
+			return exportConfig(name, idxPrefix, beatVersion)
+		}),
 	}
 }
+
+func exportConfig(name, idxPrefix, beatVersion string) error {
+	b, err := instance.NewBeat(name, idxPrefix, beatVersion)
+	if err != nil {
+		return fmt.Errorf("error initializing beat: %s", err)
```

Comment:
I know this was there before but I don't like that this and the next error message are the same. Makes it harder to find where the error came from.

Reply:
Actually we could centralize this pattern in 1 place in the cmd.

Reply:
And and "error creating beat" and "error initializing beats"